### PR TITLE
:bug: make analysis runs cancellable

### LINKED ISSUE #131: User should be able to cancel an analysis
Currently, when the user chooses to run analysis the "process" is not cancellable. This makes it so that when something goes wrong, the only recourse is for the user to restart vscode extension host.

### PR DESCRIPTION
**Pull request by djzager**
Fixes #131

### CODE REVIEW

File: vscode/src/client/analyzerClient.ts [hunk 1]
```diff
@@ -253,10 +253,16 @@ export class AnalyzerClient {
             )}`,
           );
 
-          const response: any = await this.rpcConnection!.sendRequest(
-            "analysis_engine.Analyze",
-            requestParams,
-          );
+          const cancellationPromise = new Promise((_, reject) => {
+            token.onCancellationRequested(() => {
+              reject(new Error("Operation canceled by the user."));
```
Comment (rszwajko) [inline vscode/src/client/analyzerClient.ts:258]: 1. rejecting is equivalent to throwing an exception -  I cannot test it but the `catch` block should get called. Consider resolving and exiting in a controlled way.
2. in the vscode code base  `token.isCancellationRequested` check is used sometimes to support the callback (I guess just-in-case the action would be cancelled before the callback was added)

Comment (djzager) [reply vscode/src/client/analyzerClient.ts:264]: Good catch. Addressed both.

File: vscode/src/client/analyzerClient.ts [hunk 2]
```diff
@@ -279,10 +279,22 @@ export class AnalyzerClient {
...
+            return;
+          }
+
+          const cancellationPromise = new Promise((_, reject) => {
+            token.onCancellationRequested(() => {
+              reject(new Error("Operation canceled by the user."));
+            });
+          });
+
+          const response: any = await Promise.race([
```
Comment (rszwajko) [inline vscode/src/client/analyzerClient.ts:294]: I'd rather resolve the promise i.e.:
```
const {response, isCancelled } = await Promise.race(....
```
and branch based on the returned flag.

Comment (rszwajko) [reply vscode/src/client/analyzerClient.ts]: @djzager 
I've added a commit with the fix to this PR

**Review by ibolton336 — APPROVED** [review]

**Review by sjd78 — APPROVED** [review]
So technically the analysis process won't actually get cancelled, but the UI will be unblocked.

Good enough for now.

**Pull request merged**